ios/FluentUI/Vnext/Core/Theming.swift

```diff
@@ -51,23 +51,80 @@ class MSFTokensBase {
     static var defaultValue: FluentUIStyle {
         return FluentUIThemeManager.S
     }
 }
 
+struct WindowProviderKey: EnvironmentKey {
+    static var defaultValue: FluentUIWindowProvider? {
+        return nil
+    }
+}
+
 extension EnvironmentValues {
     var theme: FluentUIStyle {
         get { self[ThemeKey.self] }
         set { self[ThemeKey.self] = newValue }
     }
+
+    var windowProvider: FluentUIWindowProvider? {
+        get { self[WindowProviderKey.self] }
+        set { self[WindowProviderKey.self] = newValue }
+    }
 }
 
 extension View {
-    func usingTheme(_ theme: FluentUIStyle) -> some View {
+
+    /// Overrides a theme for a specific SwiftUI View and its view hierarchy.
+    /// - Parameter theme: Instance of the custom overriding theme.
+    /// - Returns: The view with its theme environment value overriden.
+    func customTheme(_ theme: FluentUIStyle) -> some View {
         environment(\.theme, theme)
     }
+
+    /// Sets the window provider used to compute the theme associated with the SwiftUI View
+    /// based on the window that it belongs to.
+    /// - Parameter windowProvider: An instance of an class that implements the FluentUIWindowProvider protocol.
+    /// - Returns: The view with the window provider set for itself and its view hierarchy.
+    func windowProvider(_ windowProvider: FluentUIWindowProvider?) -> some View {
+        environment(\.windowProvider, windowProvider)
+    }
+
+    /// SwiftUI Views using this modifier need to define the following environment values:
+    ///  - @Environment(\.theme) var theme: FluentUIStyle
+    ///  - @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    ///
+    /// The instances of these environment value properties will be passed to the SwiftUI view's subviews in its hierarchy.
+    ///
+    /// The logic behind the order or precedence to retrieve the correct theme to be used is:
+    ///  1. If the SwiftUI View gets a non-default theme instance through the environment value property,
+    ///    it will be used to override the theme for this View and all the sub Views in its hierarchy.
+    ///
+    ///  2. If the theme property retrieve is the default theme instance, the windowProvider will be used to
+    ///  retrieve the theme associated with the window that this View belongs to.
+    ///
+    ///  3. If the windowProvider is nil the default theme property will be used.
+    ///
+    /// - Parameters:
+    ///   - tokens: An instance of a subclass of MSFTokensBase that defines the design tokens used in that particular SwiftUI View.
+    ///   - theme: theme property (environment value) of the SwiftUI View.
+    ///   - windowProvider: windowProvider property  (environment value) of the SwiftUI View that will potentially retrieve the theme to be applied.
+    /// - Returns: The resulting View either with the computed theme applied to its design tokens.
+    func designTokens(_ tokens: MSFTokensBase,
```

Comment:
should we add this in `ViewModifier` as opposed to `View`

https://developer.apple.com/documentation/swiftui/viewmodifier

Reply:
To be honest I don't know the benefit of doing this as we still need the extension as below to apply the modifier, but I'm adhering to Apple's recommendation.

Thanks for the suggestion!

```
extension View {
    func borderedCaption() -> some View {
        modifier(BorderedCaption())
    }
}
```